Sources/XCTest/XCTestCase.swift

```diff
@@ -308,4 +308,62 @@ extension XCTestCase {
             completionHandler(error)
         }
     }
+    
+    /// Creates and returns an expectation for a notification.
+    ///
+    /// - Parameter notificationName: The name of the notification the
+    ///   expectation observes for.
+    /// - Parameter object: The object whose notifications the expectation will
+    ///   receive; that is, only notifications with this object are observed by
+    ///   the test case. If you pass nil, the expectation doesn't use
+    ///   a notification's object to decide whether it is fulfilled.
+    /// - Parameter handler: If provided, the handler will be invoked when the
+    ///   notification is observed. It will not be invoked on timeout. Use the
+    ///   handler to further investigate if the notification fulfills the 
+    ///   expectation.
+    ///
+    /// - Note: The function used to observe the notification 
+    ///   (addObserverForName(_ name: String?, object obj: AnyObject?, 
+    ///   queue queue: NSOperationQueue?, 
+    ///   usingBlock block: (NSNotification) -> Void) -> NSObjectProtocol) 
+    ///   allows a nil notification name for its criteria of observing
+    ///   notifications, the Objective-C implementation of this function 
+    ///   requires a notification name.
```

Comment:
Is this `Note` important to an end user? I think it could be removed. Perhaps it could be moved to an `// Inline comment` below, where we call `. addObserverForName()`?